### PR TITLE
Simplify frame idx

### PR DESCRIPTION
**Pull request by tdaede**

### CODE REVIEW

File: src/api.rs [hunk 1]
```diff
@@ -795,25 +795,20 @@ impl<T: Pixel> ContextInner<T> {
...
+    }
 
-      if !self.needs_more_frames(self.frame_invariants[&idx].number) {
-        self.idx += 1;
-        return Err(EncoderStatus::LimitReached);
-      }
-      idx
-    };
+    if !self.needs_more_frames(self.frame_invariants[&self.idx].number) {
+      self.idx += 1;
```
Comment (lu-zero) [inline src/api.rs]: If we hit the limit we should not have anything more to output and `idx`, `frames_processed`, `segment_start_*` and `frame_count` could be reset before continuing.

(note to me: `needs_more_frames` should be renamed if we keep using it here)

Comment (tdaede) [reply src/api.rs]: Er, this particular line of code is not checking for the limit, is it? Also, why would we reset those?

Comment (lu-zero) [reply src/api.rs]: It does, and that's why I'd rename that function:
``` rust
  !(self.limit == 0 || self.frame_invariants[&self.idx].number < self.limit)
```

Comment (tdaede) [reply src/api.rs]: Bleh. Do you know why we are even incrementing self.idx at all here then?

Comment (lu-zero) [reply src/api.rs]: Because originally it was returning a NeedsMoreFrames and when I changed it to avoid an infiniloop I forgot to remove it ^^;

Comment (tdaede) [reply src/api.rs]: OK, I dropped it. Seems to still pass tests.

**Pull request merged**